cmd/controller/main.go

```diff
@@ -262,6 +269,48 @@ func main() {
 	})
 }
 
+func watchCertsDirectory(httpsServer *https.Server, logger *logrus.Entry) {
+	cancel, err := fswatch.Watch(logger, certDir, time.Second*5, func() {
+		logger.Info("Certificate files changed. Reloading...")
+
+		newCert, err := readCertFile()
+		if err != nil {
+			logger.WithError(err).Error("Failed to reload certificates")
+			return
+		}
+
+		// Create a new http.Server with the updated TLS configuration
+		newHTTPServer := &http.Server{
+			Addr:    ":8081",
+			Handler: httpsServer.Mux,
+		}
+		newHTTPServer.TLSConfig = &tls.Config{
+			Certificates: []tls.Certificate{*newCert},
+		}
+
+		// Update the TLS configuration
+		go func() {
+			if err := newHTTPServer.ListenAndServeTLS("", ""); err != nil {
+				logger.WithError(err).Error("Failed to update TLS configuration")
+			}
+		}()
```

Comment:
Hi @markmandel and @zmerlynn,
The moto is to update reloaded certificates in the existing server. But, however i did not find a way to update `httpsServer.TLSConfig`. I thought of creating a new server and updating the reloaded certificates. But it doesn't seems correct to me and technically it is wrong. Since, we are using custom server by utilizing this package https://github.com/googleforgames/agones/blob/main/pkg/util/https/server.go. Can i get right approach or suggestions on this?

Reply:
I found a great StackOverflow post on this, with links to several examples:
https://stackoverflow.com/questions/37473201/is-there-a-way-to-update-the-tls-certificates-in-a-net-http-server-without-any-d

TL;DR - https://pkg.go.dev/crypto/tls#Config exposes a `GetCertificate(...)`  function that will be called when a certificate is required - returning whatever is the current certificate from that is the appropriate way to go.

Reply:
Sure, Thanks I will check this.

Reply:
This is the package we should update:
https://github.com/googleforgames/agones/blob/1708fe9a6efdad4061c7a53ec8835dc92d4f4fbe/pkg/util/https/server.go